remote.rb

```diff
@@ -3,24 +3,139 @@
 require 'pry'
 require 'json'
 
-browser = Watir::Browser.new
-browser.goto('file:///home/zik/ruby/web.html')
+def trans_array (array, hash, date_arr, index)
+  array.each do |element| 
+    if element.css('td font tt')[4].text.strip.gsub(/\"/,'').match(/^\d{2}\/\d{2}\/\d{2}\s*/)
+      d = element.css('td font tt')[4].text.strip.gsub(/\"/,'')
+      post_d = Date.strptime(d, '%d/%m/%y').to_json
+    else 
+      d = date_arr[index]
+      post_d = Date.strptime(d, '%d/%m/%y').to_json
+      index += 1
+    end
+
+    if element.css('td font tt').size == 7
+      comission_element = element.css('td font tt')[5].text.strip.gsub(/\"/,'')
+      total_element     = element.css('td font tt')[6].text.strip.gsub(/\"/,'')
+    elsif element.css('td font tt').size == 6
+      comission_element = element.css('td font tt')[4].text.strip.gsub(/\"/,'')
+      total_element     = element.css('td font tt')[5].text.strip.gsub(/\"/,'')
+    end
+
+    hash.push(
+      trans_date:   Date.strptime(element.css('td font tt')[0].text.strip.gsub(/\"/,''), '%d/%m/%y').to_json,
+      details:      element.css('td font tt')[1].text.strip.gsub(/\"/,''),
+      amount:       element.css('td font tt')[2].text.strip.gsub(/\"/,''),
+      currency:     element.css('td font tt')[3].text.strip.gsub(/\"/,''),
+      post_date:    post_d,
+      comission:    comission_element,
+      total_amount: total_element
+    )
+  end
+  return hash
```

Comment:
function always return last executed line. no need to add return. you can leave just `hash`.